diff --git a/AutocompleteUnstyled/useAutocomplete.js b/AutocompleteUnstyled/useAutocomplete.js
index 8bb17682cf79bf201ab080833d10194832e6080e..53357d7cbf8fc9a6eca319f68af9f0fc8854129b 100644
--- a/AutocompleteUnstyled/useAutocomplete.js
+++ b/AutocompleteUnstyled/useAutocomplete.js
@@ -344,6 +344,10 @@ export default function useAutocomplete(props) {
     const getNextIndex = () => {
       const maxIndex = filteredOptions.length - 1;
 
+      if (diff === 'clear') {
+        return -1;
+      }
+
       if (diff === 'reset') {
         return defaultHighlighted;
       }
@@ -1009,6 +1013,9 @@ export default function useAutocomplete(props) {
       onMouseDown: event => {
         // Prevent blur
         event.preventDefault();
+      },
+      onMouseLeave: () => {
+        changeHighlightedIndex({ diff: 'clear' });
       }
     }),
     getOptionProps: ({
