diff --git a/components/panel/GridPanel.js b/components/panel/GridPanel.js
index e832b6b806fa0e0a40b79ba299917c81198daaf1..5c0551f98dfe7dc57bd7b168f8d1eb4d9f7e34d2 100644
--- a/components/panel/GridPanel.js
+++ b/components/panel/GridPanel.js
@@ -76,7 +76,7 @@ const GridPanel = /*#__PURE__*/React.forwardRef((props, ref) => {
 
   return /*#__PURE__*/_jsx(GridPanelRoot, _extends({
     ref: ref,
-    placement: "bottom-start",
+    placement: "top-start",
     className: clsx(className, classes.panel),
     anchorEl: anchorEl,
     modifiers: modifiers
